### PR TITLE
Web3 integration

### PR DESCRIPTION
**Pull request by blockchainFull2022**
add wallet connect.
seed minting functionality.

### CODE REVIEW

File: app/page.tsx [hunk 1]
```diff
@@ -2,17 +2,26 @@
...
+import {whiteList} from '@/ui/hooks/whitelist';
 
 export default function Page() {
   const [open, setOpen] = useState(false);
   const handleOpen = () => setOpen(!open);
+  const walletProvider: any = useContext(AppContext);
+  const { whitelist, setWhitelist} = walletProvider;
+  
+  useEffect(() => {
+    console.log("whitelist:", whiteList)
```
Comment (HexSleeves) [inline app/page.tsx]: remove console.log for final build

File: app/page.tsx [hunk 2]
```diff
@@ -2,17 +2,26 @@
...
 import degen from '@/public/images/img_degen.png';
 import SocialBar from '@/ui/bars/SocialBar';
 import MintDialog from '@/ui/dialog/MintDialog';
+import {AppContext} from '@/ui/hooks/AccountProvider';
+import {whiteList} from '@/ui/hooks/whitelist';
 
 export default function Page() {
   const [open, setOpen] = useState(false);
   const handleOpen = () => setOpen(!open);
+  const walletProvider: any = useContext(AppContext);
```
Comment (HexSleeves) [inline app/page.tsx]: can we provide typings instead of any?

File: ui/bars/Navbar.tsx [hunk 3]
```diff
@@ -2,14 +2,78 @@
...
+    network: "goerli",
+    theme: "light",
+    cacheProvider: false,
+    providerOptions: {
+    },
+  });
+
+  const connectWallet = async () => {
+    try {
+      console.log("connect wallet")
```
Comment (HexSleeves) [inline ui/bars/Navbar.tsx]: tear down

File: ui/bars/Navbar.tsx [hunk 4]
```diff
@@ -2,14 +2,78 @@
...
+      const epsContract = new ethers.Contract(epsAddress, epsAbi, ethersProvider);
+      const epsAddresses = await epsContract.getAddresses(web3Accounts[0], passAddress, 1, true, true);
+      setColdwallets(epsAddresses);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  const disConnectWallet = async () => {
+    console.log("disconnect")
```
Comment (HexSleeves) [inline ui/bars/Navbar.tsx]: tear down

File: ui/dialog/MintDialog.tsx [hunk 5]
```diff
@@ -36,20 +43,90 @@ const DialogFooterButton = ({ children, ...props }: Omit<ButtonProps, 'ref'>) =>
...
+  const [mintColdWallet, setMintColdWallet] = useState<string>();
+
+  const walletProvider: any = useContext(AppContext);
+  const { state, account, setAccount, library, setLibrary, provider, setProvider, loading, setLoading, whitelist, coldwallets} = walletProvider;
+
+
+  useEffect(() => {
+    ( async () => {
+    if ( coldwallets && whitelist ) {
+      console.log("cold wallets:", coldwallets)
```
Comment (HexSleeves) [inline ui/dialog/MintDialog.tsx]: tear down

File: ui/dialog/MintDialog.tsx [hunk 6]
```diff
@@ -36,20 +43,90 @@ const DialogFooterButton = ({ children, ...props }: Omit<ButtonProps, 'ref'>) =>
...
-
     const value = Number(e.target.value);
-    setMintedDegens(clampMintAmount(value));
+    if(value <= mintedMaxDegens ) {
+      setMintedDegens(clampMintAmount(value));
+    }
   }, []);
 
+  const onMint = async () => {
+    console.log("mint wallet:", mintColdWallet, mintedMaxDegens)
```
Comment (HexSleeves) [inline ui/dialog/MintDialog.tsx]: tear

File: ui/dialog/MintDialog.tsx [hunk 7]
```diff
@@ -101,14 +178,13 @@ const MintDialog = ({ open, handleOpen }: MintDialogProps) => {
...
-        >
-          Mint
-        </DialogFooterButton>
+        {
+          loading ?
+          <DialogFooterButton color="green"> Minting </DialogFooterButton>
+          :
+          <DialogFooterButton color="green" onClick={onMint}> Mint </DialogFooterButton>
+        }
+
```
Comment (HexSleeves) [inline ui/dialog/MintDialog.tsx]: we could use tailwind to add some loading overlays or cool animations https://tailwindcss.com/docs/animation

**Review by lecoqjacob — APPROVED** [review]

**Comment by lecoqjacob** [comment]
By `tear` or `tear down` i mean just remove the console logs before the final push please

File: ui/dialog/MintDialog.tsx [hunk 8]
```diff
@@ -36,20 +43,90 @@ const DialogFooterButton = ({ children, ...props }: Omit<ButtonProps, 'ref'>) =>
...
+      if (res.transactionHash) {
+        setLoading(false)
+        alert("mint success!")
+      }
+    } catch (err) {
+      console.log(err)
+      setLoading(false)
+    }
+  } 
+
```
Comment (HexSleeves) [inline ui/dialog/MintDialog.tsx]: @blockchainFull2022 dont use any type please

@soulofmischief check this logic. Good bit of this is just foreign to me

File: ui/dialog/MintDialog.tsx [hunk 9]
```diff
@@ -36,20 +43,90 @@ const DialogFooterButton = ({ children, ...props }: Omit<ButtonProps, 'ref'>) =>
...
+            mintWallet = coldWallet;           
+          }
+        }
+      }
+
+      setMintedMaxDegens(mintAmount);
+      setMintColdWallet(mintWallet);
+    }
+    })();
+  }, [coldwallets, whitelist])
```
Comment (HexSleeves) [inline ui/dialog/MintDialog.tsx]: @soulofmischief check this logic

File: ui/dialog/MintDialog.tsx (hunk 8 above)
Comment (HexSleeves) [inline ui/dialog/MintDialog.tsx]: @soulofmischief check this logic as a second pass

**Review by lecoqjacob — CHANGES_REQUESTED** [review]

**Comment by lecoqjacob** [comment]
@soulofmischief  i also cannot view vercel for some reason so I cant see why the deploy failed

**Pull request merged**